src/global.h

```diff
@@ -146,6 +146,16 @@ LED bar:      lbr
 #define AUD_MIX_FADER_MAX                100
 #define AUD_MIX_PAN_MAX                  100
 
+// coefficient for averaging channel levels for automatic fader adjustment
+#define AUTO_FADER_ADJUST_ALPHA          0.2f
+
+// target level for auto fader adjustment in decibels
```

Comment:
I'm wondering if we're not going to want to make this configurable per client.  Is -30dB based on the number of other clients summed levels not clipping?  Say I had twice the number of clients or half the number of clients, am I going to want a different target?

Reply:
I have chosen -30dB as the weakest channel we can support for adjustment. It does _not_ address the number of channels. I have chosen -30dB as a compromise between weaker channels we still want to support (seeing that the level meter value is at the lower half for some choir members)  and the loss of signal strength (and resulting quantization). 

The following table explains my thoughts about the minimum supported channel level:

Channel level | Attenuation | Comment
------------- | ------------ | -----------
0dB | -30dB | Maximum input signal gain
-10dB | -20dB
... | ...
**-30dB** | 0dB | Weakest input signal being supported, fader at its maximum (0dB)
-40dB | (+10dB) | Cannot be compensated, will still be weaker

We could also think about using -35dB as the target level since the faders range from -35dB to 0 dB.

The target level for preventing clipping would be `20 * log10(1. / cntChannels)`, e.g., -20dB for 20 channels or approximately -30dB for 32 channels. However, I think this would only apply when each channel is at its maximum level, which will seldom be the case in practice.

We could also think about setting the target level dynamically: We would take the minimum of the channel levels (weakest channel) and the required attenuation such that we do not get clipping.

Reply:
@pljones 
I have now committed a version which takes groups into account, prevents clipping by mixing and limits the overall attenuation (the minimum slider value is -35dB). I have run some tests with test vectors in a separate file (not committed) to verify the implementation. However, I did not yet have the chance to test this version with multiple channels in a real environment.

Reply:
OK... I _think_ I understand.